src/shop-cart-item.js

```diff
@@ -219,11 +222,15 @@ class ShopCartItem extends Element {
 
   _setCartItem(quantity) {
     this.dispatchEvent(new CustomEvent('set-cart-item', {
-      bubbles: true, composed: true, detail: {
+      bubbles: true, composed: true, detail: { quantity }}));
+    store.dispatch({
+      type: 'set-cart-item',
+      detail: {
```

Comment:
Consider structuring actions after the [Flux Standard Action proposal](https://github.com/acdlite/flux-standard-action). This structure is recognized by a lot of third party supporting libraries.